pkg/operator/controller/dns/controller.go

```diff
@@ -145,6 +147,13 @@ func (r *reconciler) publishRecordToZones(zones []configv1.DNSZone, record *iov1
 	result := reconcile.Result{}
 	var statuses []iov1.DNSZoneStatus
 	for i := range zones {
+		// Only publish the record if the DNSRecord has been modified
```

Comment:
@Miciah can the TODO in L143-146 be removed as part of this change?

Reply:
Yup!  Fixed.  Thanks!